src/utils/LogsUtil.ts

```diff
@@ -205,13 +205,13 @@ export async function compareWithLocalLogs(logs: Array<Log>) {
                 localLogs[i].codetime_metrics.keystrokes > 100 ||
                 localLogs[i].codetime_metrics.lines_added > 0)
         ) {
-            localLogs[i].milestones = await fetchMilestonesByDate(localLogs[i].date);
+            localLogs[i].milestones = await fetchMilestones(localLogs[i].date);
         }
     }
 
     if (localLogs.length < logs.length) {
         for (let i = localLogs.length; i < logs.length; i++) {
```

Comment:
I would recommend getting these logs sorted if they're not already sorted, and using the oldest and latest date to fetch everything without looping